src/test/java/edition/academy/seventh/security/AuthenticationRestControllerTestIT.java

```diff
@@ -211,4 +217,140 @@ public void should_returnForbiddenStatus_when_userAlreadyInDatabaseWithNotAllowe
+                            .header("Origin", "http://localhost:4200"))
+            .andExpect(status().isOk())
+            .andExpect(content().string(containsString("User registered successfully!")));
+    // Then
+    assertEquals(Set.of(roleRepository.findByName(RoleName.ROLE_ADMIN).get()), userRepository.findByUsername("newAdminByAdmin").get().getRoles());
+  }
+
+  @Test
+  public void should_returnForbiddenStatus_when_tryToRegisterWithAdminRoleByNotExistingAdmin() throws Exception {
+    // Given
+    this.mockMvc
+            .perform(
+                    post("/auth/sign_up")
+                            .content(
+                                    "{\n"
+                                            + "\t\"email\": \"userwannabe2@user.pl\",\n"
+                                            + "\t\"username\": \"userwannabe2\",\n"
+                                            + "\t\"role\": [\"user\"],\n"
+                                            + "\t\"password\": \"piesek123\"\n"
+                                            + "}")
+                            .contentType("application/json")
+                            .header("Origin", "http://localhost:4200"))
+            .andExpect(status().isOk())
+            .andExpect(content().string(containsString("User registered successfully!")));
+    String body =
+            this.mockMvc
+                    .perform(
+                            post("/auth/sign_in")
+                                    .content(
+                                            "{\n"
+                                                    + "\t\"username\": \"userwannabe2\",\n"
+                                                    + "\t\"password\": \"piesek123\"\n"
+                                                    + "}")
+                                    .contentType("application/json"))
+                    .andExpect(status().isOk())
+                    .andReturn()
+                    .getResponse()
+                    .getContentAsString();
+
+    JSONObject jsonObject = new JSONObject(body);
+    String token = jsonObject.getString("accessToken");
+    // When
+    this.mockMvc
+            .perform(
+                    post("/newUser").header("Authorization", "Bearer " + token)
+                            .content(
+                                    "{\n"
+                                            + "\t\"email\": \"newAdminByAdmin@admin.pl\",\n"
+                                            + "\t\"username\": \"newAdminByAdmin\",\n"
+                                            + "\t\"role\": [\"admin\"],\n"
+                                            + "\t\"password\": \"admin12\"\n"
+                                            + "}")
+                            .contentType("application/json")
+                            .header("Origin", "http://localhost:4200"))
+    /*// Then*/        .andExpect(status().isForbidden());
```

Comment:
Przenieś to może na linę pod //When (wcześniej tak było rozwiązane z //Given i // When)

Reply:
ustalone